packages/client/src/index.ts

```diff
@@ -12,23 +11,44 @@ components.CounterTable.update$.subscribe((update) => {
   );
 });
 
-// Create a World contract instance
-const signer = network.signer.get();
-if (!signer) throw new Error("No signer");
-
-const worldContract = IWorld__factory.connect(
-  networkConfig.worldAddress,
-  signer
-);
-
 // Just for demonstration purposes: we create a global function that can be
 // called to invoke the Increment system contract via the world. (See IncrementSystem.sol.)
-(window as any).increment = async () => {
-  const txResult = await worldContract.mud_increment_increment({
-    gasLimit: 1_000_000,
-    gasPrice: 0,
+(window as any).incrementFast = async () => {
+  const time = Date.now();
+
+  const fastTx = fastTxExecute(worldContract, "mud_increment_increment", [
+    { gasLimit: 300_000 },
+  ]);
+
+  fastTx
+    .then((t) => {
+      console.log("fastTx", t);
+      console.log("fastTx time till sending tx", Date.now() - time, "ms");
+      return t.tx;
+    })
+    .then(async (t) => {
+      console.log("t", t);
+      const result = await t.wait();
+      console.log("fastTx result", result);
+      console.log("fastTx time till result", Date.now() - time, "ms");
+    });
+};
+
+(window as any).incrementRegular = async () => {
+  const time = Date.now();
+  const regularTx = worldContract.mud_increment_increment({
+    gasLimit: 300_000,
   });
 
-  console.log(txResult);
-  console.log(await txResult.wait());
+  regularTx
+    .then((t) => {
+      console.log("regularTx", t);
+      console.log("regularTx time till sending tx", Date.now() - time, "ms");
+      return t;
+    })
+    .then(async (t) => {
```

Comment:
I remember seeing that mixing `.then` with `async` callbacks leads to hard-to-catch promise rejections and generally not advised.

You could write this as `.then((tx) => tx.wait(0)).then((result) => ...);`